HeterogeneousCore/CUDAUtilities/src/CachingDeviceAllocator.h

```diff
@@ -653,8 +655,9 @@ namespace notcub {
       cudaError_t error = cudaSuccess;
       int entrypoint_device = INVALID_DEVICE_ORDINAL;
       int current_device = INVALID_DEVICE_ORDINAL;
+      std::unique_lock<std::mutex> mutex_locker(mutex, std::defer_lock);
```

Comment:
Please add a comment along
```suggestion
      // CMS: use RAII instead of (un)locking explicitly
      std::unique_lock<std::mutex> mutex_locker(mutex, std::defer_lock);
```